### ISSUE TITLE
Debugging quality of life imporvements: --preserve-crlf and --export-mapping [FILE]

### ISSUE THREAD

**Issue by LeStahL** [laurentlb/Shader_Minifier#322]
I noticed recently that it would be very nice to have certain debugging information available, like

* A `--no-crlf-removal` flag to preserve line numbers (because I find myself using clang-format to find lines in the minified source manually a lot).

* The mapping of ids with , in a format like 
```
main="a"
main.fragColor="a.b"
trace="c"
trace.dist="c.d"
main.@anonymous.i="a.@anonymous.e"
unneccessary=""
 ...
```
maybe exposed over a command line parameter `--export-mapping file.txt`? This would simplify identifying the origin of expressions that break in the minified source.

**Comment by eldritchconundrum** [comment]
For the first point, wouldn't it be sufficient to use the existing `--format indented` flag?

**Comment by LeStahL** [comment]
> For the first point, wouldn't it be sufficient to use the existing --format indented flag?

Yes, that's exactly what I was looking for, thank you for pointing it out! (And sorry for RTFM enough) :)
